src/main/java/pl/akademiaspecjalistowit/jamfactory/controller/JamFactoryControllerAdvice.java

```diff
@@ -8,15 +8,22 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
-import pl.akademiaspecjalistowit.jamfactory.exception.JamFactoryException;
+import pl.akademiaspecjalistowit.jamfactory.exception.BusinessException;
+import pl.akademiaspecjalistowit.jamfactory.exception.ProductionException;
 
 @ControllerAdvice
 public class JamFactoryControllerAdvice {
 
-    @ExceptionHandler(JamFactoryException.class)
-    public ResponseEntity<RejectResponse> handleTravelException(JamFactoryException e) {
-        RejectResponse rejectResponse = new RejectResponse("przekraczajaca zdolnośc produkcyjna");
-        return new ResponseEntity<>(rejectResponse, HttpStatus.UNPROCESSABLE_ENTITY);
+    @ExceptionHandler(BusinessException.class)
+    public ResponseEntity<RejectResponse> handleTravelException(BusinessException e) {
+        return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY)
+                .body(new RejectResponse(e.getMessage(), ErrorCode.BUSINEES_ERROR));
+    }
+
+    @ExceptionHandler(ProductionException.class)
+    public ResponseEntity<RejectResponse> handleTravelException(ProductionException e) {
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                .body(new RejectResponse(e.getMessage(), ErrorCode.JAM_PRODUCTION_LIMIT_EXCEEDED));
```

Comment:
coś w zadaniu jest nabłytano, dla wypadka pobierania planu kiedy coś pójdzie nie tak, to zwracany HTTP 400, ale może z innym kodem, a nie JAM_PRODUCTION_LIMIT_EXCEEDED?